misc/rb.c

```diff
@@ -0,0 +1,495 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
```

Comment:
The direction from Anas is to put this stuff under lib/, misc/ is now deprecated.
The headers can still be in include/misc/ AFAIK